switchbot/devices/curtain.py

```diff
@@ -54,6 +57,18 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._settings: dict[str, Any] = {}
         self.ext_info_sum: dict[str, Any] = {}
         self.ext_info_adv: dict[str, Any] = {}
+        self._is_opening: bool = False
+        self._is_closing: bool = False
+
+    def _set_parsed_data(
+        self, advertisement: SwitchBotAdvertisement, data: dict[str, Any]
+    ) -> None:
+        """Set data."""
+        _new_data = replace(
+            advertisement, data=self._sb_adv_data.data | {"data": data}
+        )
+        self._update_motion_direction(_new_data.data["data"].get("inMotion"), self._get_adv_value("position"), _new_data.data["data"].get("position"))
```

Comment:
Why create new data when we throw it away after the next line?

Reply:
This was copy/paste from elsewhere. I wasn't exactly sure what this replace was doing if I'm being honest but it seemed like it was merging the advertisement with the existing values from _sb_adv_data. And since I am using data, I thought this might be necessary.  Happy to learn I'm wrong since you know a lot more about python than I do, I'm a C# guy by trade.

Reply:
I reread the documentation and I think you're saying I just need something more simple like this?
```
_new_data = self._sb_adv_data.data | {"data": data}
self._update_motion_direction(_new_data["data"].get("inMotion"), self._get_adv_value("position"), _new_data["data"].get("position"))
```
?